### PR TITLE
Corrected how a role with AeroHIVE devices is determined.

### PR DESCRIPTION
**Pull request by erSitzt**

### CODE REVIEW

File: lib/pf/Switch/AeroHIVE.pm [hunk 1]
```diff
@@ -239,8 +239,8 @@ sub returnRadiusAccessAccept {
     try {
 
         $logger->debug("[$mac] network device (".$self->{'_id'}.") supports roles. Evaluating role to be returned");
-        my $roleResolver = pf::roles::custom->instance();
-        my $role = $roleResolver->getRoleForNode($mac, $self);
+        my $role = $self->getRoleByName($user_role);
+
 
         # Roles are configured and the user should have one
         if (defined($role)) {
```
Comment (fdurand) [inline lib/pf/Switch/AeroHIVE.pm]: if (defined($role)  && isenabled($this->{_RoleMap})) {

....

if (isenabled($this->{_VlanMap})) {

**Comment by julsemaan** [comment]
Please remove the change you made to the admin guide as that part of the guide was completely reworked. When that part will be merged, it will prevent your code from being merged.

Thanks!

**Comment by erSitzt** [comment]
Sorry, seems i missed a notifcation for your comment.

**Comment by erSitzt** [comment]
The pull request also contains some fixes to lldp phone detection for the ProCurve 2920.
Is it ok to have this in this pull request ?

regards
Dennis

**Comment by julsemaan** [comment]
As they're both related to Switch modules it's ok for this time.

Next time split them.

FYI, that should be merged just after we release 5.3 which should happen next week

**Pull request merged**